### ISSUE TITLE
Automation Bugs

### ISSUE THREAD

**Issue by mjashanks** [Budibase/budibase#660]
- [x] The Save record Automation Creates a reacrod - we should rename Create (there is also an Update block)
- [x] The Debug/ Log block doesnt seem to do anything (maybe cos i cant see the server output anywere?)

- [x] Using the "Filter" automation block, i created the condition " {{ Rating }} (number) > Is Greater Than > 5" .... I created a record with a Rating = 10, and my automation didnt run (it previously ran ok without the filter). I suspect its because its not comparing numbers, but strings
- [ ] Save record automation doesn't show date fields, or attachments fields

- [ ] No way to reorder workflow steps - have to delete all the following steps to insert on in the middle

- [ ] "Test Automation" - im not really sure how useful this is, without being able to define the inputs to the workflow. E.g. for a record saved trigger, i'd need to be able to give it some test data

**Comment by mike12345567** [comment]
@mjashanks the "Test automation" button actually in the backend creates a fake record which is attached to the trigger, this was to simplify the user experience of testing a little and reduce the work that was needed in the front end at the time. In the future we had thought we'd put together something that could work out the correct structure of the record and allow the user to input that.

The Debug/log automation to be honest isn't really a real automation and could be removed, it was one purely for testing the async loading of automations from the repo and it only logs to the server right now.

Re-ordering of automation steps is something me and Andrew talked about, it adds some complexity because we are using the step number to state which step to pull context from, however I think it might be possible, @aptkingston do you have any thoughts on this?

The expanded types for automations, dates, attachments and soon linked records, these are each individual features that probably need to be thought about. Me and @shogunpurple talked about linked records and decided for launch it would add too much complexity to support them for automations, however dates and attachments are their own discussions.

I'll rename the save record and have a look at the filter!

**Comment by aptkingston** [comment]
I've already renamed the "Save Record" step actually on our branch - great minds think alike!

Regarding the save record action not showing certain fields - as @mike12345567 said that's deliberate because it was a new type at the time and we simplified for speed by excluding more "complicated" types like that. Part of the problem is that all fields need to support binding, which means having their value as a string, but it's not possible to have a datepicker that's happy with a value of `{{ trigger.date }}`. So adding any of these more complex types would require some thought and maybe UI redesign that allows any given field to be either entered in it's correct type, or bound to a previous step's output. That probably means either a toggle beside every field or have a custom component which actually opens a popover and contains the necessary components for either binding or entering the correct type.

As for attachments, I'm not sure if that's even something that should be in there... We would need to add support for passing around binary blobs between steps to make it useful I think. And linked records have been purposefully excluded as said above.

Step reording could be implemented, it's just something that has never been there before and so wasn't added as part of the revamp either. I agree that it would be nice to have. As Mike said, reordering would requiring inspecting all values of all step parameters and incrementing or decrementing all affected steps as required. Certainly possible and would be happy to implement this.
